examples/z_scout.c

```diff
@@ -90,12 +90,12 @@ void drop(void *context) {
 
 int main(int argc, char **argv) {
     z_init_logger();
-    z_owned_config_t config = z_config_default();
-    printf("Scouting...\n");
     int *context = malloc(sizeof(int));
     *context = 0;
+    z_owned_scouting_config_t config = z_scouting_config_default();
     z_owned_closure_hello_t closure = z_closure(callback, drop, context);
-    z_scout(Z_ROUTER | Z_PEER, z_move(config), z_move(closure), 1000);
+    printf("Scouting...\n");
+    z_scout(z_move(config), z_move(closure));
```

Comment:
I find that passing `what` as a first parameter is kind of more direct and readable API-wise. Opinions?

Reply:
I think most people won't be extremely familiar with the concept of whatamis, and I don't think it's that important that one must familiarise with it in order to do a simple scout